packages/@okta/vuepress-site/docs/guides/authenticators-google-authenticator/main/java/enrollmentintegrationsteps.md

```diff
@@ -0,0 +1,99 @@
+// make sure it is persisted for web applications
+proceedContext = authenticationResponse.getProceedContext();
+
+// get the authenticationStatus
+AuthenticationStatus authenticationStatus = authenticationResponse.getAuthenticationStatus();
+
+switch (authenticationStatus) {
+    case SUCCESS:
+        … your code …
+    case PASSWORD_EXPIRED:
+        … your code …
+}
+```
+
+### 4: Select a possible authenticator factor
+
+If you configured your Okta org correctly, you need to respond to two specific authenticator statuses to handle this scenario in addition to `SUCCESS` and `PASSWORD_EXPIRED`:
+
+* `AWAITING_PROFILE_ENROLLMENT` that is covered in this section
+* `AWAITING_AUTHENTICATOR_ENROLLMENT_SELECTION` that is covered in [the challenge flow section](#integrate-sdk-for-authenticator-challenge).
+
+The names of the authenticators available for enrollment or challenge can be found in the `AuthenticationResponse` object's `authenticators` collection. Display a list of authenticators to select the Google Authenticator for enrollment.
+
+> **Note:** The `isChallengeFlow` session variable is set to `false` if the user needs to enroll Google Authenticator first, and `true` if they have already done so.
+
+```java
+case AWAITING_AUTHENTICATOR_ENROLLMENT_SELECTION:
+
+    // There are multiple potential factors, for this example we are going to
+    // assume Google Authenticator (OTP) is used
+    Authenticator authenticator = authenticationResponse.getAuthenticators().stream()
+            .filter(auth -> auth.getType().equals("otp"))
+            .findFirst() // find the first "otp" factor or throw an exception
+            .orElseThrow(() -> {
+                String availableFactors = authenticationResponse.getAuthenticators().stream()
+                        .map(auth -> auth.getLabel() + " - " + auth.getType())
+                        .collect(Collectors.joining(", "));
+                return new RuntimeException("OTP factor not found, existing options: " + availableFactors);
+            });
+
+    // Tell Okta which factor was selected, and go back into the state machine with the new response
+    authenticationResponse = idxAuthenticationWrapper.selectAuthenticator(proceedContext, authenticator);
+```
+
+### 5: Retrieve shared secret and QR Code
+
+The next `AuthenticationResponse` object will contain information about the OTP factor and will contain a QR Code and shared secret.
+
+> **NOTE:** the _shared secret_ is typically used when a QR code cannot be displayed or scanned.
+
+Display the QR code and/or shared secret to the user, so they can register the factor with Google Authenticator.
+
+```java
+authenticationResponse.getContextualData().getSharedSecret()
+authenticationResponse.getContextualData().getQrcode()
+```
+
+Prompt the user for a OTP code, and continue the authentication flow:
+
+```java
+// send the code to Okta, and go back into the state machine with the new response
```

Comment:
// Send the code to Okta and handle the next response.

Reply:
```suggestion
// Send the code to Okta and handle the next response.
```